modules/operators/spec/map-response.spec.ts

```diff
@@ -0,0 +1,79 @@
+import { noop, Observable, of, throwError } from 'rxjs';
+import { mapResponse } from '..';
+import { concatMap, finalize } from 'rxjs/operators';
+
+describe('mapResponse', () => {
+  it('should invoke next callback on next', () => {
+    const nextCallback = jest.fn<void, [number]>();
+
+    of(1, 2, 3)
+      .pipe(
+        mapResponse({
+          next: nextCallback,
+          error: noop,
+        })
+      )
+      .subscribe();
+
+    expect(nextCallback.mock.calls).toEqual([[1], [2], [3]]);
+  });
+
+  it('should invoke error callback on error', () => {
+    const errorCallback = jest.fn<void, [{ message: string }]>();
+    const error = { message: 'error' };
+
+    throwError(() => error)
+      .pipe(
+        mapResponse({
+          next: noop,
+          error: errorCallback,
+        })
+      )
+      .subscribe();
+
+    expect(errorCallback).toHaveBeenCalledWith(error);
+  });
+
+  it('should invoke error callback on the exception thrown in next', () => {
+    const errorCallback = jest.fn<void, [{ message: string }]>();
+    const error = { message: 'error' };
+
+    function producesError() {
+      throw error;
+    }
+
+    of(1)
+      .pipe(
+        mapResponse({
+          next: producesError,
+          error: errorCallback,
+        })
+      )
+      .subscribe();
+
+    expect(errorCallback).toHaveBeenCalledWith(error);
+  });
```

Comment:
these three tests should be changed. Unlike `tapResponse`, where we verify that callbacks are invoked, here we should verify the result of `mapResponse`. Example:

(Note: I haven't tested this code, so it may require minor changes)

```ts
  it('should map the emitted value using the next callback', () => {
    const nextCallback = jest.fn<void, [number]>();

    of(1, 2, 3)
      .pipe(
        mapResponse({
          next: (num) => num + 1,
          error: noop,
        })
      )
      .subscribe(nextCallback);

    expect(nextCallback.mock.calls).toEqual([[2], [3], [4]]);
  });

  it('should map thrown error using the error callback', () => {
    const errorCallback = jest.fn<void, [string]>();
    const error = 'error message';

    throwError(() => error)
      .pipe(
        mapResponse({
          next: noop,
          error: (error: string) => `mapped ${error}`,
        })
      )
      .subscribe({ error: errorCallback });

    expect(errorCallback).toHaveBeenCalledWith('mapped error message');
  });
```